haystack/modeling/model/optimization.py

```diff
@@ -119,20 +96,15 @@ def initialize_optimizer(
     :param distributed: Whether training on distributed machines
     :param grad_acc_steps: Number of steps to accumulate gradients for. Helpful to mimic large batch_sizes on small machines.
     :param local_rank: rank of the machine in a distributed setting
-    :param use_amp: Optimization level of nvidia's automatic mixed precision (AMP). The higher the level, the faster the model.
-                    Options:
-                    "O0" (Normal FP32 training)
-                    "O1" (Mixed Precision => Recommended)
-                    "O2" (Almost FP16)
-                    "O3" (Pure FP16).
-                    See details on: https://nvidia.github.io/apex/amp.html
+    :param use_amp: Whether to use automatic mixed precision (AMP) natively implemented in PyTorch.
+                    Find more information at https://pytorch.org/docs/stable/amp.html
```

Comment:
In addition to the `logger.warning` in the function does it make sense to update the docstring to explain that `use_amp` is no longer used by this function?

Reply:
Yes, please. 👍